src/pyedb/dotnet/edb_core/edb_data/raptor_x_simulation_setup_data.py

```diff
@@ -0,0 +1,448 @@
+        super().__init__(pedb, edb_object)
+        self._setup_type = "kRaptorX"
+        self._edb_setup_info = None
+
+    @pyedb_function_handler
+    def create(self, name=None):
+        """Create an HFSS setup."""
+        self._name = name
+        self._create(name)
+        return self
+
+    @property
+    def setup_type(self):
+        return self._setup_type
+
+    @property
+    def settings(self):
+        return RaptorXSimulationSettings(self._edb_setup_info)
+
+    @property
+    def enabled(self):
+        return self.settings.enabled
+
+    @enabled.setter
+    def enabled(self, value):
+        self.settings.enabled = value
+
+    @property
+    def position(self):
+        return self._edb_setup_info.Position
+
+    @position.setter
+    def position(self, value):
+        if isinstance(value, int):
+            self._edb_setup_info.Position = value
+
+    @property
+    def frequency_sweeps(self):
+        return list(self._edb_setup_info.SweepDataList)
+
+    @pyedb_function_handler()
+    def add_frequency_sweep(self, name=None, frequency_sweep=None):
+        """Add frequency sweep.
+
+        Parameters
+        ----------
+        name : str, optional
+            Name of the frequency sweep.
+        frequency_sweep : list, optional
+            List of frequency points.
+
+        Returns
+        -------
+        :class:`pyedb.dotnet.edb_core.edb_data.hfss_simulation_setup_data.EdbFrequencySweep`
+
+        Examples
+        --------
+        >>> setup1 = edbapp.create_hfss_setup("setup1")
+        >>> setup1.add_frequency_sweep(frequency_sweep=[
+        ...                           ["linear count", "0", "1kHz", 1],
+        ...                           ["log scale", "1kHz", "0.1GHz", 10],
+        ...                           ["linear scale", "0.1GHz", "10GHz", "0.1GHz"],
+        ...                           ])
+        """
+        if name in self.frequency_sweeps:
+            return False
+        if not name:
+            name = generate_unique_name("sweep")
+        return EdbFrequencySweep(self, frequency_sweep, name)
+
+
+class RaptorXSimulationSettings(object):
+    def __init__(self, edb_setup_info):
+        self._edb_setup_info = edb_setup_info
+        self._simulation_settings = edb_setup_info.SimulationSettings
+        self._general_settings = RaptorXGeneralSettings(self._edb_setup_info)
+        self._advanced_settings = RaptorXSimulationAdvancedSettings(self._edb_setup_info)
+        self._simulation_settings = self._edb_setup_info.SimulationSettings
+
+    @property
+    def general_settings(self):
+        return self._general_settings
+
+    @property
+    def advanced_settings(self):
+        return self._advanced_settings
+
+    @property
+    def enabled(self):
+        return self._simulation_settings.Enabled
+
+    @enabled.setter
+    def enabled(self, value):
+        if isinstance(value, bool):
+            self._simulation_settings.Enabled = value
```

Comment:
Can you log an error if value is not bool ?

Reply:
Done